src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -517,6 +517,58 @@ def _update_source_sample_period(self, now: datetime) -> bool:
 
         return True
 
+    def _update_buffer_len(self) -> bool:
+        """Update the length of the buffer based on the source properties.
+
+        Returns:
+            Whether the buffer length was changed (was really updated).
+        """
+        input_sampling_period_s = self._source_properties.sampling_period_s
+
+        # To make type checking happy
+        assert input_sampling_period_s is not None
+        assert self._buffer.maxlen is not None
+
+        config = self._config
+
+        # If we are upsampling, one sample could be enough for back-filling, but
+        # we store max_data_age_in_periods for input periods, so resampling
+        # functions can do more complex inter/extrapolation if they need to.
+        if input_sampling_period_s > config.resampling_period_s:
+            new_buffer_len = input_sampling_period_s * config.max_data_age_in_periods
+        # If we are upsampling, we want a buffer that can hold
+        # max_data_age_in_periods * resampling_period_s seconds of data, and we
+        # one sample every input_sampling_period_s.
+        else:
+            new_buffer_len = (
+                config.resampling_period_s
+                / input_sampling_period_s
+                * config.max_data_age_in_periods
+            )
+
+        new_buffer_len = max(1, math.ceil(new_buffer_len))
+        if new_buffer_len > config.max_buffer_len:
+            _logger.warning(
+                "The new buffer length (%s) for timeseries %s is too big, using %s instead",
+                new_buffer_len,
+                self._name,
+                config.max_buffer_len,
+            )
```

Comment:
Shouldn't it be an error?
And the exception should be raised.

Reply:
I don't think an exception should be raised because we don't get anything from that, there isn't much to do by the caller if this happens (except maybe rebuild the whole thing increasing the `max_buffer_len` or adjusting the `resampling_period_s`, but those are more deployment things IMHO), but I could turn it into an error. It is not really fatal, but it is true that it can degrade the quality of the resampling if it happens so maybe it is better to shout louder.

Does this look acceptable?

Reply:
Oh, I see you found a bug, my intention was to truncate the buffer to `max_buffer_len`, which I now see it's not being done! I will fix that too, and should probably add a test too.

Reply:
Fixed (without raising an exception for now)